codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydotnet/ServiceCodegen.java

```diff
@@ -689,6 +690,16 @@ private String getConstraints(
     };
   }
 
+  public String capitalizeSegment(final String segment) {
```

Comment:
Given underscores can appear in any Smithy namespaces/identifiers by the looks of it (https://smithy.io/2.0/spec/model.html#shape-id-abnf), we might hit this same error in several other places in the future. It would be a lot better to push this fix into `capitalizeNamespaceSegment` instead, and to split on `_` and handle each piece the same, instead of assuming we should give up and use `StringUtils.capitalize` instead.

Reply:
Done